connector/github/github.go

```diff
@@ -244,23 +244,23 @@ func (c *githubConnector) HandleCallback(s connector.Scopes, r *http.Request) (i
 		EmailVerified: true,
 	}
 
-	if s.Groups {
-		var groups []string
-		if len(c.orgs) > 0 {
-			if groups, err = c.listGroups(ctx, client, user.Login); err != nil {
-				return identity, err
-			}
-		} else if c.org != "" {
-			inOrg, err := c.userInOrg(ctx, client, user.Login, c.org)
-			if err != nil {
-				return identity, err
-			}
-			if !inOrg {
-				return identity, fmt.Errorf("github: user %q not a member of org %q", user.Login, c.org)
-			}
-			if groups, err = c.teams(ctx, client, c.org); err != nil {
-				return identity, fmt.Errorf("github: get teams: %v", err)
-			}
+	// Org and team membership should always be checked if specified by config.
+	if len(c.orgs) > 0 {
+		// Error if groups scope is not provided by the client; the groups scope and
+		// orgs must be used simultaneously. API endpoints might return a 404
+		// response otherwise.
+		if !s.Groups {
+			return identity, errors.New("github: orgs are specified without 'groups' scope provided")
```

Comment:
But API requests won't succeed in this case because we don't set the [`read:org` scope](https://github.com/coreos/dex/blob/master/connector/github/github.go#L138-L141).